subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/ImplementationChanges.java

```diff
@@ -50,42 +49,34 @@ public boolean accept(ChangeVisitor visitor) {
                 executable.getDisplayName(), previousImplementation.getTypeName(), currentImplementation.getTypeName()));
         }
         if (currentImplementation.isUnknown()) {
```

Comment:
Does it make sense to keep track of an unknown implementation anymore if all we do is throw an exception down the line?

Reply:
If we need the fact that the implementation is unknown, we could use an `Optional` to keep track of that, too. But my hunch is that we can get rid of tracking unknown completely.